### DISCUSSION TITLE
can i use global variables at next13 api route function?

### DISCUSSION THREAD

**Discussion by Matheus8174** [vercel/next.js#55014] [Help]
### Summary

I've been working in match making like system and use a global variable to store the second client id, the first will come from a queue. As you can see the prevClient is a global variable, as next route works in a serverless way i don't know if it's a good approach have this global variable but it's working in the development.

```typescript
type MessageShape = { data: string }

let prevClient: MessageShape;

export const chatQueue = Queue<MessageShape>(
  "api/queues/chat",
  async job => {

    console.log(job);
    const prevData = prevClient?.data;

    if(prevData === job.data) return;

    if(!prevData) {
      prevClient = job;

      return;
    };

    if(prevData) {
      console.log([prevData, job.data])

      const chatId = randomUUID();

      const clients = [
        rest.channels.get(`client:${prevData}`).publish('chat-id', chatId),
        rest.channels.get(`client:${job.data}`).publish('chat-id', chatId)
      ];

      await Promise.all(clients);
    }

    prevClient = undefined
  }
)

export const POST = chatQueue;
```

**Comment by devjiwonchoi** [comment]
I don't see any problem with it, but why would it be necessary?

**Reply by Matheus8174** [reply]
because to create a Chat i need two clients and i get it from the queue, i got one and save it in this variable the next one that comes from the queue is matched with this from the variable, and i create a chat id for both. Basically i need this variable to store a client because i can just got one at a time in the queue callback.

**Reply by devjiwonchoi** [reply]
I see. Based on my understanding, it won't mess up since the handler functions will be called by the request method so I'm assuming the variable won't be mutated.

One suggestion is to group up the codes that handle the clients and pass them to the handlers.

**Reply by Matheus8174** [reply]
ok i will use this global variable approach so. When you talk about group up the code you say create a function like this

```typescript
import queueJobHandle, { type Job } from './QueueJobHandle';

export const chatQueue = Queue<Job>('api/queues/chat', queueJobHandle);

export const POST = chatQueue;
```

**Reply by devjiwonchoi** [reply]
Right, that way you don't need to worry about breaking the process:)
